crypto/x509/x509_vfy.c

```diff
@@ -3054,6 +3047,10 @@ static int build_chain(X509_STORE_CTX *ctx)
             }
             curr = sk_X509_value(ctx->chain, i - 1);
 
+            /* Stop with success if the current cert is trusted */
```

Comment:
A pity that Stephen Henson did not document the rationale for that behavior.

As stated several times before, the very idea of this PR is to stop chain building as soon as possible, making the resulting chain as short as possible.
This has several advantages:
* IMO more security because only a minimal number of certs are involved in the chain, which less chances of errors/bug/attacks in between.
* efficiency
* sometimes it simply does not make sense to try and find an issuer first, and doing so can cause needless spurious errors